src/DotLiquid/StandardFilters.cs

```diff
@@ -313,42 +300,43 @@ public static string Lstrip(string input)
         /// <summary>
         /// Strip all trailing whitespace from input
         /// </summary>
-        /// <param name="input"></param>
-        /// <returns></returns>
+        /// <param name="input">Input to be transformed by this filter</param>
         public static string Rstrip(string input)
         {
             return input?.TrimEnd();
         }
 
         /// <summary>
-        /// Converts the input object into a formatted currency as specified by the culture info.
+        /// Converts the input object into a formatted currency as specified by the context cuture, or languageTag parameter (if provided).
         /// </summary>
-        /// <param name="input"></param>
-        /// <param name="cultureInfo"></param>
-        /// <returns></returns>
-        public static string Currency(object input, string cultureInfo = null)
+        /// <remarks>
+        /// If the input is a string it is ALWAYS parsed using the context culture, the optional languageTag parameter is only applied for rendering.
+        /// </remarks>
+        /// <param name="context">default source of culture information</param>
+        /// <param name="input">value to be parsed and formatted as a Currency</param>
+        /// <param name="languageTag">optional override language for rendering, for example 'fr-FR'</param>
+        /// <seealso href="https://shopify.dev/api/liquid/filters/money-filters#money" >Shopify Money filter</seealso>
+        public static string Currency(Context context, object input, string languageTag = null)
         {
-
-            if (decimal.TryParse(input.ToString(), out decimal amount))
+            var culture = context.CurrentCulture;
+            if (languageTag != null) //Allow an empty string, which represent the InvariantCulture
             {
-                if (cultureInfo.IsNullOrWhiteSpace())
-                {
-                    cultureInfo = CultureInfo.CurrentCulture.Name;
-                }
-
-                var culture = new CultureInfo(cultureInfo);
+                culture = new CultureInfo(languageTag.Trim());

```

Comment:
Doing this condition directly on the culture variable initialization with a ternary assignment would follow an immutable coding pattern for clarity.

Reply:
Done